frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2896,10 +2901,40 @@ inferRefMaybeConstFormals(ResolutionContext* rc,
   return result;
 }
 
+static std::vector<std::tuple<const Decl*, QualifiedType>>
+collectGenericFormals(Context* context, const TypedFnSignature* tfs) {
+  std::vector<std::tuple<const Decl*, QualifiedType>> ret;
+
+  // Skip the 'this' formal since it will always be generic if one of the
+  // "real" formals is generic.
+  int formalIdx = 0;
+  if (tfs->untyped()->formalName(0) == USTR("this")) {
```

Comment:
I think you could use ``isMethod`` here instead.

Reply:
I think I just moved this code, but I will make the change.